File: optimal-settings-backend/src/models/game_settings.rs
use serde::{Deserialize, Serialize};
use serde_with::skip_serializing_none;

#[skip_serializing_none]
#[derive(Serialize, Deserialize)]
pub struct GameSettings {
    pub high: Option<Vec<[String; 2]>>,
    pub medium: Option<Vec<[String; 2]>>,
    pub low: Option<Vec<[String; 2]>>,
    pub additional_informations: Option<String>,
}

File: optimal-settings-frontend/src/app/games/types/game-settings.ts
import { z } from "zod";

export const GameSettings = z.object({
  high: z.array(z.tuple([z.string(), z.string()])).optional(),
  medium: z.array(z.tuple([z.string(), z.string()])).optional(),
  low: z.array(z.tuple([z.string(), z.string()])).optional(),
  additionalInformations: z.string().optional(),
});

export type GameSettings = z.infer<typeof GameSettings>;
